recipes-devtools/riscv-tools/files/0002-Makefile.in-Don-t-specify-a-march-if-one-isn-t-speci.patch

```diff
@@ -0,0 +1,39 @@
+From 4c0651d47ef53edf262c26e486543acb2b353cfb Mon Sep 17 00:00:00 2001
+From: Alistair Francis <alistair.francis@wdc.com>
+Date: Fri, 2 Nov 2018 10:22:09 -0700
+Subject: [PATCH] Makefile.in: Don't specify a march if one isn't specified
+
+This breaks the build when using the OE toolchain.
+
+Signed-off-by: Alistair Francis <alistair.francis@wdc.com>
+---
+ Makefile.in | 13 -------------
+ 1 file changed, 13 deletions(-)
+
+diff --git a/Makefile.in b/Makefile.in
+index aeabdc2..ac696fb 100644
+--- a/Makefile.in
++++ b/Makefile.in
+@@ -50,19 +50,6 @@ else
+   project_ver:=@PACKAGE_VERSION@
+ endif
+ 
+-# If --with-arch is not specified, it defaults to whatever the compiler's
+-# default is. The -with-abi is not necessary for this project. Unconditionally
+-# compile it with a no-float ABI. i.e., ilp32 for 32-bit and lp64 for 64-bit.
+-
+-ifeq (@WITH_ARCH@,)
+-  march := -march=$(shell @CC@ -v 2>&1 | \
+-                    sed 's/ /\n/g' | grep with-arch | awk -F= '{print $$2}')
+-else
+-  march := -march=@WITH_ARCH@
+-endif
+-is_32bit := $(findstring 32,$(march))
+-mabi := -mabi=$(if $(is_32bit),ilp32,lp64)
```

Comment:
is there a way to make it upstreamable way ?

Reply:
Not that I can think of. We have to ensure it's not set and in the FreedomSDK environment they need it to be set.